pantheon/src/main/java/tech/pegasys/pantheon/cli/operator/OperatorSubCommand.java

```diff
@@ -0,0 +1,322 @@
+      LOG.info("Importing public keys from configuration.");
+      JsonArray keys = nodesConfig.getJsonArray("keys");
+      keys.stream().forEach(this::importPublicKey);
+    }
+
+    /**
+     * Imports a single public key.
+     *
+     * @param publicKeyObject The public key.
+     */
+    private void importPublicKey(final Object publicKeyObject) {
+      try {
+        final SECP256K1.PublicKey publicKey =
+            SECP256K1.PublicKey.create(BytesValue.fromHexString((String) publicKeyObject));
+        writeKeypair(publicKey, null);
+        LOG.info("Public key imported from configuration.({})", publicKey.toString());
+      } catch (IOException e) {
+        LOG.error("An error occurred while trying to import node public key.", e);
+      }
+    }
+
+    /** Generates nodes keypairs. */
+    private void generateNodesKeys() {
+      final int nodesCount = nodesConfig.getInteger("count");
+      LOG.info("Generating {} nodes keys.", nodesCount);
+      IntStream.range(0, nodesCount).forEach(this::generateNodeKeypair);
+    }
+
+    /**
+     * Generate a keypair for a node.
+     *
+     * @param node The number of the node.
+     */
+    private void generateNodeKeypair(final int node) {
+      try {
+        LOG.info("Generating keypair for node {}.", node);
+        final SECP256K1.KeyPair keyPair = SECP256K1.KeyPair.generate();
+        writeKeypair(keyPair.getPublicKey(), keyPair.getPrivateKey());
+
+      } catch (IOException e) {
+        LOG.error("An error occurred while trying to generate node keypair.", e);
+      }
+    }
+
+    /**
+     * Writes public and private keys in separate files. Both are written in the same directory
+     * named with the address derived from the public key.
+     *
+     * @param publicKey The public key.
+     * @param privateKey The private key. No file is created if privateKey is NULL.
+     * @throws IOException
+     */
+    private void writeKeypair(
+        final SECP256K1.PublicKey publicKey, final SECP256K1.PrivateKey privateKey)
+        throws IOException {
+      final Address nodeAddress = Util.publicKeyToAddress(publicKey);
+      addressesForGenesisExtraData.add(nodeAddress);
+      final Path nodeDirectoryPath = keysDirectory.resolve(nodeAddress.toString());
+      Files.createDirectory(nodeDirectoryPath);
+      createFileAndWrite(nodeDirectoryPath, publicKeyFileName, publicKey.toString());
+      if (privateKey != null) {
+        createFileAndWrite(nodeDirectoryPath, privateKeyFileName, privateKey.toString());
+      }
+    }
+
+    /**
+     * Computes RLP encoded exta data from pre filled list of addresses.
+     *
+     * @throws IOException
+     */
+    private void processExtraData() throws IOException {
+      final JsonGenesisConfigOptions genesisConfigOptions =
+          JsonGenesisConfigOptions.fromJsonObject(genesisConfig.getJsonObject("config"));
+      if (genesisConfigOptions.isIbft2() || genesisConfigOptions.isIbftLegacy()) {
```

Comment:
There are actually 2 different `IbftExtraData` classes - one for ibft2 and one for the legacy ibft algorithm.  So this should just check `isIbft2()` for now (or check against all consensus fields and use the right utility).